src/common_subset.rs

```diff
@@ -102,41 +102,41 @@ pub struct CommonSubset<NodeUid: Rand> {
     decided: bool,
 }
 
+pub type CommonSubsetStep<NodeUid> = Step<NodeUid, BTreeMap<NodeUid, ProposedValue>>;
```

Comment:
In the above files, you were using `<N>` (not `<NodeUid>`) as the type parameter. Sticking to one naming convention might be beneficial.